// @tailwindcss/container-queries/dist/index.js
"use strict";
var _plugin = /*#__PURE__*/ _interopRequireDefault(require("tailwindcss/plugin"));
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
module.exports = (0, _plugin.default)(function containerQueries(param) {
    var matchUtilities = param.matchUtilities, matchVariant = param.matchVariant, theme = param.theme;
    var parseValue = function parseValue(value) {
        var ref;
        var ref1;
        var numericValue = (ref1 = (ref = value.match(/^(\d+\.\d+|\d+|\.\d+)\D+/)) === null || ref === void 0 ? void 0 : ref[1]) !== null && ref1 !== void 0 ? ref1 : null;
        if (numericValue === null) return null;
        return parseFloat(value);
    };
    var ref;
    var values = (ref = theme("containers")) !== null && ref !== void 0 ? ref : {};
    matchUtilities({
        "@container": function(value, param) {
            var modifier = param.modifier;
            return {
                "container-type": value,
                "container-name": modifier
            };
        }
    }, {
        values: {
            DEFAULT: "inline-size",
            normal: "normal"
        },
        modifiers: "any"
    });
    matchVariant("@", function() {
        var value = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : "", modifier = (arguments.length > 1 ? arguments[1] : void 0).modifier;
        var parsed = parseValue(value);
        return parsed !== null ? "@container ".concat(modifier !== null && modifier !== void 0 ? modifier : "", " (min-width: ").concat(value, ")") : [];
    }, {
        values: values,
        sort: function sort(aVariant, zVariant) {
            var a = parseFloat(aVariant.value);
            var z = parseFloat(zVariant.value);
            if (a === null || z === null) return 0;
            // Sort values themselves regardless of unit
            if (a - z !== 0) return a - z;
            var _modifier;
            var aLabel = (_modifier = aVariant.modifier) !== null && _modifier !== void 0 ? _modifier : "";
            var _modifier1;
            var zLabel = (_modifier1 = zVariant.modifier) !== null && _modifier1 !== void 0 ? _modifier1 : "";
            // Explicitly move empty labels to the end
            if (aLabel === "" && zLabel !== "") {
                return 1;
            } else if (aLabel !== "" && zLabel === "") {
                return -1;
            }
            // Sort labels alphabetically in the English locale
            // We are intentionally overriding the locale because we do not want the sort to
            // be affected by the machine's locale (be it a developer or CI environment)
            return aLabel.localeCompare(zLabel, "en", {
                numeric: true
            });
        }
    });
}, {
    theme: {
        containers: {
            xs: "20rem",
            sm: "24rem",
            md: "28rem",
            lg: "32rem",
            xl: "36rem",
            "2xl": "42rem",
            "3xl": "48rem",
            "4xl": "56rem",
            "5xl": "64rem",
            "6xl": "72rem",
            "7xl": "80rem"
        }
    }
});
